ale/base/data_naif.py

```diff
@@ -322,23 +322,77 @@ def sensor_position(self):
                 # spkezr returns a vector from the observer's location to the aberration-corrected
                 # location of the target. For more information, see:
                 # https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/FORTRAN/spicelib/spkezr.html
-                state, _ = spice.spkezr(self.target_name,
+                state, _ = spice.spkezr(self.spacecraft_name,
                                        time,
                                        self.reference_frame,
                                        self.light_time_correction,
-                                       self.spacecraft_name,)
+                                       self.target_name,)
                 pos.append(state[:3])
                 vel.append(state[3:])
             # By default, spice works in km, and the vector returned by spkezr points the opposite
             # direction to what ALE needs, so it must be multiplied by (-1)
-            self._position = [p * -1000 for p in pos]
-            self._velocity = [v * -1000 for v in vel]
+            self._position = [p * 1000 for p in pos]
+            self._velocity = [v * 1000 for v in vel]
         return self._position, self._velocity, self.ephemeris_time
 
+    @property
+    def frame_trace(self):
+        frame_codes = [self.sensor_frame_id]
+        _, frame_type, _ = spice.frinfo(frame_codes[-1])
+        frame_types = [frame_type]
+
+        while(frame_codes[-1] != 1):
+            try:
+                center, frame_type, frame_type_id = spice.frinfo(frame_codes[-1])
+            except Exception as e:
+                print(e)
+                break
+
+            if frame_type is 1 or frame_type is 2:
+                frame_code = 1
+
+            elif frame_type is 3:
+                matrix, frame_code, found = spice.ckfrot(frame_type_id, self.center_ephemeris_time)
+                if not found:
+                    raise Exception(f"The ck rotation from frame {frame_codes[-1]} can not \
+                                      be found due to no pointing available at requested time \
+                                      or a problem with the frame")
+            elif frame_type is 4:
+                matrix, frame_code, found = spice.tkfram(frame_type_id)
+                if not found:
+                    raise Exception(f"The tk rotation from frame {frame_codes[-1]} can not \
+                                      be found")
+            elif frame_type is 5:
+                matrix, frame_code = spice.zzdynrot(frame_type_id, center, self.center_ephemeris_time)
```

Comment:
I am not a huge fan of computing all of these rotations right here and then immediately throwing the rotation away to be recomputed later.

Reply:
Is it possible to bake this into the generation of the frame chain?